core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -528,3 +523,26 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
   private val DATE_FORMAT_YEAR_MONTH = "yyyy-MM"
   private val DATE_FORMAT_YEAR_MONTH_DAY_HOUR = "yyyy-MM-dd-HH"
 }
+
+/**
+ * Finds the full dot-separated path to a field and the data type of the field. This unifies
+ * handling of nested and non-nested fields, and allows pattern matching on the data type.
+ */
+object ExtractBaseColumn {
+  def unapply(e: Expression): Option[(String, DataType)] = e match {
+    case AttributeReference(name, dataType, _, _) =>
+      Some((escapeName(name), dataType))
+    case g: GetStructField => g.child match {
+      case ExtractBaseColumn(name, _) =>
+        Some((Seq(name, escapeName(g.extractFieldName)).mkString("."), g.dataType))
```

Comment:
do we have a test to verify the recursive call?

Reply:
Added a test with multiple levels of nesting